experiments/PlaywrightSharp/PlaywrightSharp/ApiSurface/IBrowserContext.cs

```diff
@@ -0,0 +1,411 @@
+		/// <param name="callback">Callback function that will be called in the Playwright's context.</param>
+		/// <param name="handle">
+		/// Whether to pass the argument as a handle, instead of passing by value. When passing
+		/// a handle, only one argument is supported. When passing by value, multiple arguments
+		/// are supported.
+		/// </param>
+		Task ExposeBindingAsync(string name, Action callback, bool handle);
+	
+		/// <summary>
+		/// <para>
+		/// The method adds a function called <paramref name="name"/> on the <c>window</c> object
+		/// of every frame in every page in the context. When called, the function executes
+		/// <paramref name="callback"/> and returns a [Promise] which resolves to the return
+		/// value of <paramref name="callback"/>.
+		/// </para>
+		/// <para>If the <paramref name="callback"/> returns a [Promise], it will be awaited.</para>
+		/// <para>See <see cref="IPage.ExposeFunctionAsync"/> for page-only version.</para>
+		/// <para>An example of adding an <c>md5</c> function to all pages in the context:</para>
+		/// </summary>
+		/// <param name="name">Name of the function on the window object.</param>
+		/// <param name="callback">Callback function that will be called in the Playwright's context.</param>
+		Task ExposeFunctionAsync(string name, Action callback);
+	
+		/// <summary>
+		/// <para>
+		/// Grants specified permissions to the browser context. Only grants corresponding permissions
+		/// to the given origin if specified.
+		/// </para>
+		/// </summary>
+		/// <param name="permissions">
+		/// A permission or an array of permissions to grant. Permissions can be one of the
+		/// following values:
+		/// <list type="bullet">
+		/// <item><description>`'geolocation'`</description></item>
+		/// <item><description>`'midi'`</description></item>
+		/// <item><description>`'midi-sysex'` (system-exclusive midi)</description></item>
+		/// <item><description>`'notifications'`</description></item>
+		/// <item><description>`'push'`</description></item>
+		/// <item><description>`'camera'`</description></item>
+		/// <item><description>`'microphone'`</description></item>
+		/// <item><description>`'background-sync'`</description></item>
+		/// <item><description>`'ambient-light-sensor'`</description></item>
+		/// <item><description>`'accelerometer'`</description></item>
+		/// <item><description>`'gyroscope'`</description></item>
+		/// <item><description>`'magnetometer'`</description></item>
+		/// <item><description>`'accessibility-events'`</description></item>
+		/// <item><description>`'clipboard-read'`</description></item>
+		/// <item><description>`'clipboard-write'`</description></item>
+		/// <item><description>`'payment-handler'`</description></item>
+		/// </list>
+		/// </param>
+		/// <param name="origin">The [origin] to grant permissions to, e.g. "https://example.com".</param>
+		Task GrantPermissionsAsync(string[] permissions, string origin);
```

Comment:
I created an enum for these permissions.

Reply:
Hm, yeah, the values are specified in the documentation, but not as an enum. This *might* be because we want to keep some flexibility in here and not make it a part of the API, but I'm not 100% sure. Still, while this is the case, let's keep the _generated_ method like this, and, as previously stated, we can add an overload/extension method to make the user's life simpler.